state-synchronizer/src/request_manager.rs

```diff
@@ -254,17 +259,29 @@ impl RequestManager {
                 .expect("missing network sender for peer");
             let peer_id = peer.peer_id();
             let send_result = sender.send_to(peer_id, msg.clone());
-            let result_label = if send_result.is_err() {
-                failed_peer_sends.push((peer.clone(), send_result));
+            let curr_log = log.clone().peer(peer.clone());
+            let result_label = if let Err(e) = send_result {
+                failed_peer_sends.push(peer.clone());
+                error!(curr_log
+                    .event(LogEvent::NetworkSendError)
+                    .error(&format_err!("{:?}", e)));
```

Comment:
If it's a well known type you can use that, and set the schema to print the debug version.  Might be a little cleaner / easier

In Networking, I couldn't do this, because there were like 3-4 different error types.

Reply:
should fix that in networking or make a single unified type that they can all Into into